monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,462 @@
+
+        d[self.image] = self._apply(image, guidance)
+        return d
+
+
+class FindDiscrepancyRegionsd(Transform):
+    """
+    Find discrepancy between prediction and actual during click interactions during training.
+
+    If batched is true:
+        label is in shape (B, C, D, H, W) or (B, C, H, W)
+        pred has same shape as label
+        discrepancy will have shape (B, 2, C, D, H, W) or (B, 2, C, H, W)
+
+    Args:
+        label: key to label source.
+        pred: key to prediction source.
+        discrepancy: key to store discrepancies found between label and prediction.
+        batched: whether input is batched or not.
+    """
+
+    def __init__(
+        self, label: str = "label", pred: str = "pred", discrepancy: str = "discrepancy", batched: bool = True
+    ):
+        self.label = label
+        self.pred = pred
+        self.discrepancy = discrepancy
+        self.batched = batched
+
+    @staticmethod
+    def disparity(label, pred):
+        label = (label > 0.5).astype(np.float32)
+        pred = (pred > 0.5).astype(np.float32)
+        disparity = label - pred
+
+        pos_disparity = (disparity > 0).astype(np.float32)
+        neg_disparity = (disparity < 0).astype(np.float32)
+        return [pos_disparity, neg_disparity]
+
+    def _apply(self, label, pred):
+        if not self.batched:
+            return self.disparity(label, pred)
+
+        disparity = []
+        for la, pr in zip(label, pred):
+            disparity.append(self.disparity(la, pr))
+        return disparity
+
+    def __call__(self, data):
+        d = dict(data)
+        label = d[self.label]
+        pred = d[self.pred]
+
+        d[self.discrepancy] = self._apply(label, pred)
+        return d
+
+
+class AddRandomGuidanced(Randomizable, Transform):
+    """
+    Add random guidance based on discrepancies that were found between label and prediction.
+
+    If batched is True:
+
+        Guidance is of shape (B, 2, N, # of dim) where B is batch size, 2 means positive and negative,
+        N means how many guidance points, # of dim is the total number of dimensions of the image
+        (for example if the image is CDHW, then # of dim would be 4).
+
+        Discrepancy is of shape (B, 2, C, D, H, W) or (B, 2, C, H, W)
+
+        Probability is of shape (B,)
+
+    Args:
+        guidance: key to guidance source.
+        discrepancy: key that represents discrepancies found between label and prediction.
+        probability: key that represents click/interaction probability.
+        batched: whether input is batched or not.
+    """
+
+    def __init__(
+        self,
+        guidance: str = "guidance",
+        discrepancy: str = "discrepancy",
+        probability: str = "probability",
+        batched: bool = True,
+    ):
+        self.guidance = guidance
+        self.discrepancy = discrepancy
+        self.probability = probability
+        self.batched = batched
+
+    def randomize(self, data=None):
```

Comment:
could you follow this pattern for the randomizable: https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/intensity/array.py#L69-L77 calling self.R within this method

Reply:
I am not sure how to do this.
Should we move all the logic (find_guidance, add_guidance) inside randomize?